core/src/main/scala/kafka/tools/ConsoleConsumer.scala

```diff
@@ -460,46 +466,31 @@ class DefaultMessageFormatter extends MessageFormatter {
   var printKey = false
   var printValue = true
   var printPartition = false
-  var keySeparator = "\t".getBytes(StandardCharsets.UTF_8)
-  var lineSeparator = "\n".getBytes(StandardCharsets.UTF_8)
+  var printOffset = false
+  var printHeaders = false
+  var keySeparator = utfBytes("\t")
+  var lineSeparator = utfBytes("\n")
+  var headersSeparator = utfBytes(",")
 
   var keyDeserializer: Option[Deserializer[_]] = None
   var valueDeserializer: Option[Deserializer[_]] = None
+  var headersDeserializer: Option[Deserializer[_]] = None
 
   override def init(props: Properties): Unit = {
```

Comment:
@badaiaqrandista FYI, the `init` method has been deprecated as part of KIP-597: https://github.com/apache/kafka/pull/8604. You will have to adapt your PR to support it.